Index: git/arch/arm/boot/dts/imx6q-roj.dts
===================================================================
--- git.orig/arch/arm/boot/dts/imx6q-roj.dts	2017-02-09 14:57:07.611919275 +0100
+++ git/arch/arm/boot/dts/imx6q-roj.dts	2017-02-09 17:51:14.987930849 +0100
@@ -19,48 +19,65 @@
 	model = "ROJ iMX6 Smarc board";
 	compatible = "fsl,imx6q-roj", "fsl,imx6q";
 
-        sound {
-                compatible = "fsl,imx6q-sabrelite-sgtl5000",
-                             "fsl,imx-audio-sgtl5000";
-                model = "imx6q-sabrelite-sgtl5000";
-                ssi-controller = <&ssi1>;
-                audio-codec = <&codec>;
-                audio-routing =
-                        "MIC_IN", "Mic Jack",
-                        "Mic Jack", "Mic Bias",
-                        "Headphone Jack", "HP_OUT";
-                mux-int-port = <1>;
-                mux-ext-port = <3>;
-        };
+	sound {
+		compatible = "fsl,imx6q-sabrelite-sgtl5000",
+									"fsl,imx-audio-sgtl5000";
+		model = "imx6q-sabrelite-sgtl5000";
+		ssi-controller = <&ssi1>;
+		audio-codec = <&codec>;
+		audio-routing =
+						"MIC_IN", "Mic Jack",
+						"Mic Jack", "Mic Bias",
+						"Headphone Jack", "HP_OUT";
+		mux-int-port = <1>;
+		mux-ext-port = <3>;
+	};
 
 };
 
 &cpu0 {
-       arm-supply = <&sw3_reg>;
-       soc-supply = <&sw1_reg>;
-       pu-supply = <&pu_dummy>; /* use pu_dummy if VDDSOC share with VDDPU */
+	arm-supply = <&sw3_reg>;
+	soc-supply = <&sw1_reg>;
+/*	pu-supply = <&pu_dummy>; *//* use pu_dummy if VDDSOC share with VDDPU */
+	pu-supply  = <&reg_pu>;
+
+	operating-points = <
+		/* kHz    uV */
+		996000  1252125
+		852000  1252125
+		792000  1275750
+		396000  992250
+	>;
+	fsl,soc-operating-points = <
+		/* ARM kHz  SOC-PU uV */
+		996000        1252125
+		852000        1252125
+		792000        1181250
+		396000        1181250
+	>;
+
 };
 
 &i2c1 {
-        codec: sgtl5000@0a {
-                compatible = "fsl,sgtl5000";
-                reg = <0x0a>;
-                clocks = <&clks 201>;
-                VDDA-supply = <&reg_1p8v>;
-                VDDIO-supply = <&reg_3p3v>;
-        };
+	codec: sgtl5000@0a {
+		compatible = "fsl,sgtl5000";
+		reg = <0x0a>;
+		clocks = <&clks 201>;
+		VDDA-supply = <&reg_1p8v>;
+		VDDIO-supply = <&reg_3p3v>;
+	};
 };
 
 &i2c2 {
-               mt9m111@5d {
-                	compatible = "micron,mt9m111";
-                    reg = <0x5d>;
-                	clocks = <&clks 201>;
-                	clock-names = "csi_mclk";
-                	csi_id = <1>;
-                	mclk = <24000000>;
-                	mclk_source = <0>;
-               };
+	mt9m111@5d {
+		compatible = "micron,mt9m111";
+		reg = <0x5d>;
+		clocks = <&clks 201>;
+		clock-names = "csi_mclk";
+		csi_id = <1>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+	};
 };
 
 &i2c3 {
@@ -69,84 +86,84 @@
 		reg = <0x50>;
 		pagesize = <32>;
 	};
-       pcf8563: rtc@51 {
-               compatible = "nxp,pcf8563";
-               reg = <0x51>;
-       };
-
-       ltc3676: pmic@3c {
-               compatible = "lltc,ltc3676";
-               reg = <0x3c>;
-
-               regulators {
-                       sw1_reg: sw1 {
-                               regulator-min-microvolt = <779625>;
-                               regulator-max-microvolt = <1512000>;
-                               lltc,fb-voltage-divider = <187000 200000>;
-                               regulator-ramp-delay = <7000>;
-                               regulator-boot-on;
-                               regulator-always-on;
-                       };
-
-                       sw2_reg: sw2 {
-                               regulator-min-microvolt = <1031250>;
-                               regulator-max-microvolt = <2000000>;
-                               lltc,fb-voltage-divider = <270000 180000>;
-                               regulator-ramp-delay = <7000>;
-                               regulator-boot-on;
-                               regulator-always-on;
-                       };
-
-                       sw3_reg: sw3 {
-                               regulator-min-microvolt = <779625>;
-                               regulator-max-microvolt = <1512000>;
-                               lltc,fb-voltage-divider = <187000 200000>;
-                               regulator-ramp-delay = <7000>;
-                               regulator-boot-on;
-                               regulator-always-on;
-                       };
-
-                       sw4_reg: sw4 {
-                               regulator-min-microvolt = <855937>;
-                               regulator-max-microvolt = <1660000>;
-                               lltc,fb-voltage-divider = <215000 200000>;
-                               regulator-ramp-delay = <7000>;
-                               regulator-boot-on;
-                               regulator-always-on;
-                       };
-
-                       ldo1_reg: ldo1 {
-                               regulator-min-microvolt = <0>;
-                               regulator-max-microvolt = <0>;
-                               lltc,fb-voltage-divider = <0 0>;
-                               regulator-boot-on;
-                               regulator-always-on;
-                       };
-
-                       ldo2_reg: ldo2 {
-                               regulator-min-microvolt = <2972500>;
-                               regulator-max-microvolt = <2972500>;
-                               lltc,fb-voltage-divider = <0 0>;
-                               regulator-boot-on;
-                               regulator-always-on;
-                       };
-
-                       ldo3_reg: ldo3 {
-                               regulator-min-microvolt = <1800000>;
-                               regulator-max-microvolt = <1800000>;
-                               lltc,fb-voltage-divider = <0 0>;
-                               regulator-boot-on;
-                       };
-
-                       ldo4_reg: ldo4 {
-                               regulator-min-microvolt = <1207125>;
-                               regulator-max-microvolt = <1207125>;
-                               lltc,fb-voltage-divider = <0 0>;
-                               regulator-boot-on;
-                               regulator-always-on;
-                       };
-               };
-       };
+	pcf8563: rtc@51 {
+		compatible = "nxp,pcf8563";
+		reg = <0x51>;
+	};
+
+	ltc3676: pmic@3c {
+		compatible = "ltc3676";
+		reg = <0x3c>;
+
+		regulators {
+			sw1_reg: sw1 {
+				regulator-min-microvolt = <779625>;
+				regulator-max-microvolt = <1512000>;
+				lltc,fb-voltage-divider = <178000 200000>;
+				regulator-ramp-delay = <7000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw2_reg: sw2 {
+				regulator-min-microvolt = <1031250>;
+				regulator-max-microvolt = <2000000>;
+				lltc,fb-voltage-divider = <270000 180000>;
+				regulator-ramp-delay = <7000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3_reg: sw3 {
+				regulator-min-microvolt = <779625>;
+				regulator-max-microvolt = <1512000>;
+				lltc,fb-voltage-divider = <178000 200000>;
+				regulator-ramp-delay = <7000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw4_reg: sw4 {
+				regulator-min-microvolt = <855937>;
+				regulator-max-microvolt = <1660000>;
+				lltc,fb-voltage-divider = <215000 200000>;
+				regulator-ramp-delay = <7000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			ldo1_reg: ldo1 {
+				regulator-min-microvolt = <1200000>;
+				regulator-max-microvolt = <1200000>;
+				lltc,fb-voltage-divider = <0 0>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			ldo2_reg: ldo2 {
+				regulator-min-microvolt = <2972500>;
+				regulator-max-microvolt = <2972500>;
+				lltc,fb-voltage-divider = <0 0>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			ldo3_reg: ldo3 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				lltc,fb-voltage-divider = <0 0>;
+				regulator-boot-on;
+			};
+
+			ldo4_reg: ldo4 {
+				regulator-min-microvolt = <1207125>;
+				regulator-max-microvolt = <1207125>;
+				lltc,fb-voltage-divider = <0 0>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+		};
+	};
 };
 
 &mxcfb1 {
@@ -170,14 +187,23 @@
 };
 
 &mipi_csi {
-        status = "okay";
-        ipu_id = <0>;
-        csi_id = <1>;
-        v_channel = <0>;
-        lanes = <2>;
+	status = "okay";
+	ipu_id = <0>;
+	csi_id = <1>;
+	v_channel = <0>;
+	lanes = <2>;
 };
 
 &ssi1 {
-        fsl,mode = "i2s-slave";
-        status = "okay";
+	fsl,mode = "i2s-slave";
+	status = "okay";
+};
+
+&can1 {
+	status = "okay";
 };
+
+&can2 {
+	status = "okay";
+};
+
Index: git/arch/arm/boot/dts/imx6q-roj.dtsi
===================================================================
--- git.orig/arch/arm/boot/dts/imx6q-roj.dtsi	2017-02-09 14:57:07.611919275 +0100
+++ git/arch/arm/boot/dts/imx6q-roj.dtsi	2017-02-09 17:51:15.011930849 +0100
@@ -1,4 +1,3 @@
-
 /*
  * Copyright 2013-2015 Freescale Semiconductor, Inc.
  *
@@ -27,8 +26,9 @@
 			device_type = "cpu";
 			reg = <0>;
 			next-level-cache = <&L2>;
+/*
 			operating-points = <
-				/* kHz    uV */
+				/ * kHz    uV * /
 				1200000 1275000
 				996000  1250000
 				852000  1250000
@@ -36,28 +36,31 @@
 				396000  975000
 			>;
 			fsl,soc-operating-points = <
-				/* ARM kHz  SOC-PU uV */
+				/ * ARM kHz  SOC-PU uV * /
 				1200000 1275000
 				996000	1250000
 				852000	1250000
 				792000	1175000
 				396000	1175000
 			>;
+*/
 			clock-latency = <61036>; /* two CLK32 periods */
 			clocks = <&clks IMX6QDL_CLK_ARM>,
-				 <&clks IMX6QDL_CLK_PLL2_PFD2_396M>,
-				 <&clks IMX6QDL_CLK_STEP>,
-				 <&clks IMX6QDL_CLK_PLL1_SW>,
-				 <&clks IMX6QDL_CLK_PLL1_SYS>,
-				 <&clks IMX6QDL_CLK_PLL1>,
-				 <&clks IMX6QDL_PLL1_BYPASS>,
-				 <&clks IMX6QDL_PLL1_BYPASS_SRC>;
+				<&clks IMX6QDL_CLK_PLL2_PFD2_396M>,
+				<&clks IMX6QDL_CLK_STEP>,
+				<&clks IMX6QDL_CLK_PLL1_SW>,
+				<&clks IMX6QDL_CLK_PLL1_SYS>,
+				<&clks IMX6QDL_CLK_PLL1>,
+				<&clks IMX6QDL_PLL1_BYPASS>,
+				<&clks IMX6QDL_PLL1_BYPASS_SRC>;
 			clock-names = "arm", "pll2_pfd2_396m", "step",
 				      "pll1_sw", "pll1_sys", "pll1",
 				      "pll1_bypass", "pll1_bypass_src";
+/*
 			arm-supply = <&reg_arm>;
 			pu-supply = <&reg_pu>;
 			soc-supply = <&reg_soc>;
+*/
 		};
 
 		cpu@1 {
@@ -108,7 +111,7 @@
 			fsl,max_ddr_freq = <528000000>;
 		};
 
-                gpu@00130000 {
+		gpu@00130000 {
 			compatible = "fsl,imx6q-gpu";
 			reg = <0x00130000 0x4000>, <0x00134000 0x4000>,
 			      <0x02204000 0x4000>, <0x0 0x0>,
Index: git/arch/arm/boot/dts/imx6qdl-roj-base.dtsi
===================================================================
--- git.orig/arch/arm/boot/dts/imx6qdl-roj-base.dtsi	2017-02-09 14:57:07.871919275 +0100
+++ git/arch/arm/boot/dts/imx6qdl-roj-base.dtsi	2017-02-09 14:57:07.995919275 +0100
@@ -927,30 +927,30 @@
 			ranges;
 
 			crypto: caam@2100000 {
-                                compatible = "fsl,sec-v4.0";
-                                #address-cells = <1>;
-                                #size-cells = <1>;
-                                reg = <0x2100000 0x40000>;
-                                ranges = <0 0x2100000 0x40000>;
-                                interrupt-parent = <&intc>; /* interrupts = <0 92 0x4>; */
-                                clocks = <&clks IMX6QDL_CAAM_MEM>, <&clks IMX6QDL_CAAM_ACLK>, <&clks IMX6QDL_CAAM_IPG> ,<&clks IMX6QDL_CLK_EIM_SLOW>;
-                                clock-names = "caam_mem", "caam_aclk", "caam_ipg", "caam_emi_slow";
-
-                                sec_jr0: jr0@1000 {
-                                        compatible = "fsl,sec-v4.0-job-ring";
-                                        reg = <0x1000 0x1000>;
-                                        interrupt-parent = <&intc>;
-                                        interrupts = <0 105 0x4>;
-                                };
-
-                                sec_jr1: jr1@2000 {
-                                        compatible = "fsl,sec-v4.0-job-ring";
-                                        reg = <0x2000 0x1000>;
-                                        interrupt-parent = <&intc>;
-                                        interrupts = <0 106 0x4>;
-                                };
+				compatible = "fsl,sec-v4.0";
+				#address-cells = <1>;
+				#size-cells = <1>;
+				reg = <0x2100000 0x40000>;
+				ranges = <0 0x2100000 0x40000>;
+				interrupt-parent = <&intc>; /* interrupts = <0 92 0x4>; */
+				clocks = <&clks IMX6QDL_CAAM_MEM>, <&clks IMX6QDL_CAAM_ACLK>, <&clks IMX6QDL_CAAM_IPG> ,<&clks IMX6QDL_CLK_EIM_SLOW>;
+				clock-names = "caam_mem", "caam_aclk", "caam_ipg", "caam_emi_slow";
+
+				sec_jr0: jr0@1000 {
+					compatible = "fsl,sec-v4.0-job-ring";
+					reg = <0x1000 0x1000>;
+					interrupt-parent = <&intc>;
+					interrupts = <0 105 0x4>;
 				};
 
+				sec_jr1: jr1@2000 {
+					compatible = "fsl,sec-v4.0-job-ring";
+					reg = <0x2000 0x1000>;
+					interrupt-parent = <&intc>;
+					interrupts = <0 106 0x4>;
+				};
+			};
+
 			aipstz@0217c000 { /* AIPSTZ2 */
 				reg = <0x0217c000 0x4000>;
 			};
@@ -1331,14 +1331,14 @@
 				MX6QDL_PAD_SD2_DAT0__AUD4_RXD  0x130b0
 			>;
 		};
-               pinctrl_audmux_4_2: audmux-4-2 {
-                       fsl,pins = <
-                               MX6QDL_PAD_SD2_DAT2__AUD4_TXD  0x130b0
-                               MX6QDL_PAD_SD2_CMD__AUD4_RXC   0x130b0
-                               MX6QDL_PAD_SD2_CLK__AUD4_RXFS  0x130b0
-                               MX6QDL_PAD_SD2_DAT0__AUD4_RXD  0x130b0
-                       >;
-               };
+		pinctrl_audmux_4_2: audmux-4-2 {
+			fsl,pins = <
+				MX6QDL_PAD_SD2_DAT2__AUD4_TXD  0x130b0
+				MX6QDL_PAD_SD2_CMD__AUD4_RXC   0x130b0
+				MX6QDL_PAD_SD2_CLK__AUD4_RXFS  0x130b0
+				MX6QDL_PAD_SD2_DAT0__AUD4_RXD  0x130b0
+			>;
+		};
 	};
 
 	enet {
@@ -1367,7 +1367,7 @@
 		pinctrl_esai_1: esaigrp-1 {
 			fsl,pins = <
 				MX6QDL_PAD_ENET_RXD1__ESAI_TX_FS     0x1b030
-                               MX6QDL_PAD_ENET_TX_EN__ESAI_TX3_RX2  0x1b030
+				MX6QDL_PAD_ENET_TX_EN__ESAI_TX3_RX2  0x1b030
 				MX6QDL_PAD_ENET_CRS_DV__ESAI_TX_CLK  0x1b030
 				MX6QDL_PAD_ENET_TXD0__ESAI_TX4_RX1   0x1b030
 			>;
@@ -1581,8 +1581,8 @@
 
 		pinctrl_usdhc3_1_100mhz: usdhc3grp-1-100mhz { /* 100Mhz */
 			fsl,pins = <
-				MX6QDL_PAD_SD3_CMD__SD3_CMD 0x170B9
-				MX6QDL_PAD_SD3_CLK__SD3_CLK 0x100B9
+				MX6QDL_PAD_SD3_CMD__SD3_CMD    0x170B9
+				MX6QDL_PAD_SD3_CLK__SD3_CLK    0x100B9
 				MX6QDL_PAD_SD3_DAT0__SD3_DATA0 0x170B9
 				MX6QDL_PAD_SD3_DAT1__SD3_DATA1 0x170B9
 				MX6QDL_PAD_SD3_DAT2__SD3_DATA2 0x170B9
@@ -1596,8 +1596,8 @@
 
 		pinctrl_usdhc3_1_200mhz: usdhc3grp-1-200mhz { /* 200Mhz */
 			fsl,pins = <
-				MX6QDL_PAD_SD3_CMD__SD3_CMD 0x170F9
-				MX6QDL_PAD_SD3_CLK__SD3_CLK 0x100F9
+				MX6QDL_PAD_SD3_CMD__SD3_CMD    0x170F9
+				MX6QDL_PAD_SD3_CLK__SD3_CLK    0x100F9
 				MX6QDL_PAD_SD3_DAT0__SD3_DATA0 0x170F9
 				MX6QDL_PAD_SD3_DAT1__SD3_DATA1 0x170F9
 				MX6QDL_PAD_SD3_DAT2__SD3_DATA2 0x170F9
@@ -1613,54 +1613,54 @@
 	usdhc4 {
 		pinctrl_usdhc4_1: usdhc4grp-1 {
 			fsl,pins = <
-				MX6QDL_PAD_SD4_CMD__SD4_CMD     0x17059
-				MX6QDL_PAD_SD4_CLK__SD4_CLK     0x10059
-				MX6QDL_PAD_SD4_DAT0__SD4_DATA0  0x17059
-				MX6QDL_PAD_SD4_DAT1__SD4_DATA1  0x17059
-				MX6QDL_PAD_SD4_DAT2__SD4_DATA2  0x17059
-				MX6QDL_PAD_SD4_DAT3__SD4_DATA3  0x17059
-				MX6QDL_PAD_SD4_DAT4__SD4_DATA4  0x17059
-				MX6QDL_PAD_SD4_DAT5__SD4_DATA5  0x17059
-				MX6QDL_PAD_SD4_DAT6__SD4_DATA6  0x17059
-				MX6QDL_PAD_SD4_DAT7__SD4_DATA7  0x17059
+				MX6QDL_PAD_SD4_CMD__SD4_CMD      0x17059
+				MX6QDL_PAD_SD4_CLK__SD4_CLK      0x10059
+				MX6QDL_PAD_SD4_DAT0__SD4_DATA0   0x17059
+				MX6QDL_PAD_SD4_DAT1__SD4_DATA1   0x17059
+				MX6QDL_PAD_SD4_DAT2__SD4_DATA2   0x17059
+				MX6QDL_PAD_SD4_DAT3__SD4_DATA3   0x17059
+				MX6QDL_PAD_SD4_DAT4__SD4_DATA4   0x17059
+				MX6QDL_PAD_SD4_DAT5__SD4_DATA5   0x17059
+				MX6QDL_PAD_SD4_DAT6__SD4_DATA6   0x17059
+				MX6QDL_PAD_SD4_DAT7__SD4_DATA7   0x17059
 				MX6QDL_PAD_NANDF_ALE__GPIO6_IO08 0x17059
 				MX6QDL_PAD_EIM_D18__GPIO3_IO18   0x13069
 			>;
 		};
 
-                pinctrl_usdhc4_1_100mhz: usdhc4grp-1-100mhz {
-                        fsl,pins = <
-                                MX6QDL_PAD_SD4_CMD__SD4_CMD     0x170B9
-                                MX6QDL_PAD_SD4_CLK__SD4_CLK     0x100B9
-                                MX6QDL_PAD_SD4_DAT0__SD4_DATA0  0x170B9
-                                MX6QDL_PAD_SD4_DAT1__SD4_DATA1  0x170B9
-                                MX6QDL_PAD_SD4_DAT2__SD4_DATA2  0x170B9
-                                MX6QDL_PAD_SD4_DAT3__SD4_DATA3  0x170B9
-                                MX6QDL_PAD_SD4_DAT4__SD4_DATA4  0x170B9
-                                MX6QDL_PAD_SD4_DAT5__SD4_DATA5  0x170B9
-                                MX6QDL_PAD_SD4_DAT6__SD4_DATA6  0x170B9
-                                MX6QDL_PAD_SD4_DAT7__SD4_DATA7  0x170B9
-								MX6QDL_PAD_NANDF_ALE__GPIO6_IO08 0x170B9
-								MX6QDL_PAD_EIM_D18__GPIO3_IO18   0x13069
-                        >;
-                };
-
-                pinctrl_usdhc4_1_200mhz: usdhc4grp-1-200mhz {
-                        fsl,pins = <
-                                MX6QDL_PAD_SD4_CMD__SD4_CMD     0x170F9
-                                MX6QDL_PAD_SD4_CLK__SD4_CLK     0x100F9
-                                MX6QDL_PAD_SD4_DAT0__SD4_DATA0  0x170F9
-                                MX6QDL_PAD_SD4_DAT1__SD4_DATA1  0x170F9
-                                MX6QDL_PAD_SD4_DAT2__SD4_DATA2  0x170F9
-                                MX6QDL_PAD_SD4_DAT3__SD4_DATA3  0x170F9
-                                MX6QDL_PAD_SD4_DAT4__SD4_DATA4  0x170F9
-                                MX6QDL_PAD_SD4_DAT5__SD4_DATA5  0x170F9
-                                MX6QDL_PAD_SD4_DAT6__SD4_DATA6  0x170F9
-                                MX6QDL_PAD_SD4_DAT7__SD4_DATA7  0x170F9
-								MX6QDL_PAD_NANDF_ALE__GPIO6_IO08 0x170F9
-								MX6QDL_PAD_EIM_D18__GPIO3_IO18   0x13069
-                        >;
-                };
+		pinctrl_usdhc4_1_100mhz: usdhc4grp-1-100mhz {
+			fsl,pins = <
+				MX6QDL_PAD_SD4_CMD__SD4_CMD      0x170B9
+				MX6QDL_PAD_SD4_CLK__SD4_CLK      0x100B9
+				MX6QDL_PAD_SD4_DAT0__SD4_DATA0   0x170B9
+				MX6QDL_PAD_SD4_DAT1__SD4_DATA1   0x170B9
+				MX6QDL_PAD_SD4_DAT2__SD4_DATA2   0x170B9
+				MX6QDL_PAD_SD4_DAT3__SD4_DATA3   0x170B9
+				MX6QDL_PAD_SD4_DAT4__SD4_DATA4   0x170B9
+				MX6QDL_PAD_SD4_DAT5__SD4_DATA5   0x170B9
+				MX6QDL_PAD_SD4_DAT6__SD4_DATA6   0x170B9
+				MX6QDL_PAD_SD4_DAT7__SD4_DATA7   0x170B9
+				MX6QDL_PAD_NANDF_ALE__GPIO6_IO08 0x170B9
+				MX6QDL_PAD_EIM_D18__GPIO3_IO18   0x13069
+			>;
+		};
+
+		pinctrl_usdhc4_1_200mhz: usdhc4grp-1-200mhz {
+			fsl,pins = <
+				MX6QDL_PAD_SD4_CMD__SD4_CMD      0x170F9
+				MX6QDL_PAD_SD4_CLK__SD4_CLK      0x100F9
+				MX6QDL_PAD_SD4_DAT0__SD4_DATA0   0x170F9
+				MX6QDL_PAD_SD4_DAT1__SD4_DATA1   0x170F9
+				MX6QDL_PAD_SD4_DAT2__SD4_DATA2   0x170F9
+				MX6QDL_PAD_SD4_DAT3__SD4_DATA3   0x170F9
+				MX6QDL_PAD_SD4_DAT4__SD4_DATA4   0x170F9
+				MX6QDL_PAD_SD4_DAT5__SD4_DATA5   0x170F9
+				MX6QDL_PAD_SD4_DAT6__SD4_DATA6   0x170F9
+				MX6QDL_PAD_SD4_DAT7__SD4_DATA7   0x170F9
+				MX6QDL_PAD_NANDF_ALE__GPIO6_IO08 0x170F9
+				MX6QDL_PAD_EIM_D18__GPIO3_IO18   0x13069
+			>;
+		};
 
 	};
 
Index: git/arch/arm/boot/dts/imx6qdl-roj.dtsi
===================================================================
--- git.orig/arch/arm/boot/dts/imx6qdl-roj.dtsi	2017-02-09 14:57:07.943919275 +0100
+++ git/arch/arm/boot/dts/imx6qdl-roj.dtsi	2017-02-09 18:23:15.779932977 +0100
@@ -110,21 +110,22 @@
 			gpio = <&gpio2 2 GPIO_ACTIVE_LOW>;
 			enable-active-high;
 		};
-               reg_3p3v: 3p3v {
-                       compatible = "regulator-fixed";
-                       regulator-name = "3P3V";
-                       regulator-min-microvolt = <3300000>;
-                       regulator-max-microvolt = <3300000>;
-                       regulator-always-on;
-               };
-
-               reg_1p8v: 1p8v {
-                       compatible = "regulator-fixed";
-                       regulator-name = "1P8V";
-                       regulator-min-microvolt = <1800000>;
-                       regulator-max-microvolt = <1800000>;
-                       regulator-always-on;
-               };
+
+		reg_3p3v: 3p3v {
+			compatible = "regulator-fixed";
+			regulator-name = "3P3V";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-always-on;
+		};
+
+		reg_1p8v: 1p8v {
+			compatible = "regulator-fixed";
+			regulator-name = "1P8V";
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <1800000>;
+			regulator-always-on;
+		};
 	};
 
 	sound-hdmi {
@@ -150,41 +151,40 @@
 		status = "okay";
 	};
 
-
-        v4l2_cap_1 {
-                compatible = "fsl,imx6q-v4l2-capture";
-                ipu_id = <0>;
-                csi_id = <1>;
-                mclk_source = <0>;
-                status = "okay";
-        };
+	v4l2_cap_1 {
+		compatible = "fsl,imx6q-v4l2-capture";
+		ipu_id = <0>;
+		csi_id = <1>;
+		mclk_source = <0>;
+		status = "okay";
+	};
 
 	v4l2_out {
 		compatible = "fsl,mxc_v4l2_output";
 		status = "okay";
 	};
 
-        lcd@0 {
-                compatible = "fsl,lcd";
-                ipu_id = <0>;
-                disp_id = <0>;
-                default_ifmt = "RGB565";
-                pinctrl-names = "default";
-                pinctrl-0 = <&pinctrl_ipu1_1>;
-                status = "okay";
-        };
+	lcd@0 {
+		compatible = "fsl,lcd";
+		ipu_id = <0>;
+		disp_id = <0>;
+		default_ifmt = "RGB565";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_ipu1_1>;
+		status = "okay";
+	};
 
 	i2c@0 {
-               compatible = "i2c-gpio";
-               gpios = <&gpio7  13  0 /* sda */
-                       &gpio7 12 0 /* scl */
-               >;
-               i2c-gpio,sda-open-drain;
-               i2c-gpio,scl-open-drain;
-               i2c-gpio,delay-us = <5>;        /* ~100 kHz */
-               i2c-gpio,timeout-ms = <100>;
-               #address-cells = <1>;
-               #size-cells  = <0>;
+		compatible = "i2c-gpio";
+		gpios = <&gpio7  13  0 /* sda */
+						&gpio7 12 0 /* scl */
+		>;
+		i2c-gpio,sda-open-drain;
+		i2c-gpio,scl-open-drain;
+		i2c-gpio,delay-us = <5>;        /* ~100 kHz */
+		i2c-gpio,timeout-ms = <100>;
+		#address-cells = <1>;
+		#size-cells  = <0>;
 	};
 
 };
@@ -227,8 +227,22 @@
 	fsl,cpu_pupscr_sw = <0xf>;
 	fsl,cpu_pdnscr_iso2sw = <0x1>;
 	fsl,cpu_pdnscr_iso = <0x1>;
+	fsl,ldo-bypass = <1>; /* use ldo-bypass, u-boot will check it and configure */
+	fsl,wdog-reset = <2>; /* watchdog select of reset source */
+	pu-supply = <&pu_dummy>; /* ldo-bypass:use pu_dummy if VDDSOC share with VDDPU */
 };
 
+&soc {
+	&gpu {
+		pu-supply = <&pu_dummy>; /* ldo-bypass:use pu_dummy if VDDSOC share with VDDPU */
+	};
+};
+
+&vpu {
+        pu-supply = <&pu_dummy>; /* ldo-bypass:use pu_dummy if VDDSOC share with VDDPU */
+};
+
+
 &hdmi_audio {
 	status = "okay";
 };
@@ -285,7 +297,7 @@
 				MX6QDL_PAD_GPIO_2__GPIO1_IO02        0x80000000
 				MX6QDL_PAD_GPIO_6__GPIO1_IO06        0x80000000
 				MX6QDL_PAD_GPIO_17__GPIO7_IO12       0x80000000
-                MX6QDL_PAD_GPIO_18__GPIO7_IO13       0x80000000
+				MX6QDL_PAD_GPIO_18__GPIO7_IO13       0x80000000
 				MX6QDL_PAD_NANDF_D0__GPIO2_IO00      0x80000000
 				MX6QDL_PAD_NANDF_D1__GPIO2_IO01      0x80000000
 				MX6QDL_PAD_NANDF_D2__GPIO2_IO02      0x80000000
@@ -316,14 +328,11 @@
 				MX6QDL_PAD_NANDF_RB0__GPIO6_IO10     0x80000000
 				MX6QDL_PAD_NANDF_CS0__GPIO6_IO11     0x80000000
 				MX6QDL_PAD_NANDF_CS1__GPIO6_IO14     0x80000000
-                               MX6QDL_PAD_NANDF_CS2__GPIO6_IO15     0x80000000
-                               MX6QDL_PAD_NANDF_CS3__GPIO6_IO16     0x80000000
+				MX6QDL_PAD_NANDF_CS2__GPIO6_IO15     0x80000000
+				MX6QDL_PAD_NANDF_CS3__GPIO6_IO16     0x80000000
 				MX6QDL_PAD_EIM_BCLK__GPIO6_IO31      0x80000000
 				MX6QDL_PAD_SD3_RST__GPIO7_IO08       0x80000000
-				MX6QDL_PAD_GPIO_3__CCM_CLKO2    0x130b0
-			
-				
-				
+				MX6QDL_PAD_GPIO_3__CCM_CLKO2         0x130b0
 			>;
 		};
 	};
@@ -393,9 +402,9 @@
 };
 
 &pcie {
-        reset-gpio = <&gpio1 6 0>;
-		wake-up-gpio = <&gpio2 5 0>;
-		status = "okay";
+	reset-gpio = <&gpio1 6 0>;
+	wake-up-gpio = <&gpio2 5 0>;
+	status = "okay";
 };
 
 &spdif {
@@ -444,47 +453,47 @@
 };
 
 &usbh1 {
-        status = "okay";
+	status = "okay";
 };
 
 &usdhc1 {
-        pinctrl-names = "default";
-        pinctrl-0 = <&pinctrl_usdhc1_1>;
-        cd-gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
-        wp-gpios = <&gpio2 0 GPIO_ACTIVE_HIGH>;
-        no-1-8-v;
-        keep-power-in-suspend;
-        enable-sdio-wakeup;
-        status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc1_1>;
+	cd-gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
+	wp-gpios = <&gpio2 0 GPIO_ACTIVE_HIGH>;
+	no-1-8-v;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	status = "okay";
 };
 
 &usdhc3 {
-        pinctrl-names = "default", "state_100mhz", "state_200mhz";
-        pinctrl-0 = <&pinctrl_usdhc3_1>;
-        pinctrl-1 = <&pinctrl_usdhc3_1_100mhz>;
-        pinctrl-2 = <&pinctrl_usdhc3_1_200mhz>;
-        non-removable;
-        bus-width = <8>;
-        no-1-8-v;
-        keep-power-in-suspend;
-        enable-sdio-wakeup;
-        status = "okay";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc3_1>;
+	pinctrl-1 = <&pinctrl_usdhc3_1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc3_1_200mhz>;
+	non-removable;
+	bus-width = <8>;
+	no-1-8-v;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	status = "okay";
 };
 
 &usdhc4 {
-        pinctrl-names = "default", "state_100mhz", "state_200mhz";
-        pinctrl-0 = <&pinctrl_usdhc4_1>;
-        pinctrl-1 = <&pinctrl_usdhc4_1_100mhz>;
-        pinctrl-2 = <&pinctrl_usdhc4_1_200mhz>;
-		vmmc-supply = <&reg_wlreg_on>;
-        bus-width = <4>;
-        no-1-8-v;
-        non-removable;
-        keep-power-in-suspend;
-        enable-sdio-wakeup;
-        status = "okay";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc4_1>;
+	pinctrl-1 = <&pinctrl_usdhc4_1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc4_1_200mhz>;
+	vmmc-supply = <&reg_wlreg_on>;
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	status = "okay";
 
-		brcmf: bcrmf@1 {
-			compatible = "brcm,bcm4329-fmac";
-		};
+	brcmf: bcrmf@1 {
+		compatible = "brcm,bcm4329-fmac";
+	};
 };
